save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/RunExecutionController.kt

```diff
@@ -0,0 +1,193 @@
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.http.codec.json.Jackson2JsonEncoder
+import org.springframework.security.core.Authentication
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.toEntity
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+
+/**
+ * Controller for running execution
+ */
+@RestController
+@RequestMapping("/api/$v1/run")
+@Suppress("LongParameterList")
+class RunExecutionController(
+    private val projectService: ProjectService,
+    private val executionService: ExecutionService,
+    private val executionInfoStorage: ExecutionInfoStorage,
+    private val testService: TestService,
+    private val testExecutionService: TestExecutionService,
+    private val meterRegistry: MeterRegistry,
+    configProperties: ConfigProperties,
+    objectMapper: ObjectMapper,
+) {
+    private val webClientOrchestrator = WebClient.builder()
+        .baseUrl(configProperties.orchestratorUrl)
+        .codecs {
+            it.defaultCodecs().multipartCodecs().encoder(Jackson2JsonEncoder(objectMapper))
+        }
+        .build()
+    private val scheduler = Schedulers.boundedElastic()
+
+    /**
+     * @param request incoming request from frontend
+     * @param authentication
+     * @return response with ID of created [Execution]
+     */
+    @PostMapping("/trigger")
+    fun trigger(
+        @RequestBody request: RunExecutionRequest,
+        authentication: Authentication,
+    ): Mono<StringResponse> = Mono.just(request.projectCoordinates)
+        .validateAccess(authentication) { it }
+        .map {
+            executionService.createNew(
+                projectCoordinates = request.projectCoordinates,
+                testSuiteIds = request.testSuiteIds,
+                files = request.files,
+                username = authentication.username(),
+                sdk = request.sdk,
+                execCmd = request.execCmd,
+                batchSizeForAnalyzer = request.batchSizeForAnalyzer
+            )
+        }
+        .subscribeOn(Schedulers.boundedElastic())
+        .flatMap { execution ->
+            Mono.just(execution.toAcceptedResponse())
+                .doOnSuccess {
+                    asyncTrigger(execution)
+                }
+        }
+
+    /**
+     * @param executionId ID of [Execution] which needs to be copied for new execution
+     * @param authentication
+     * @return response with ID of new created [Execution]
+     */
+    @PostMapping("/reTrigger")
```

Comment:
AFAIR, we want to have kebab-case in urls instead of camelCase

Reply:
will change to re-trigger